src/lib.rs

```diff
@@ -98,6 +98,38 @@ impl Client {
         self.drive_receive(parse_ascii_response).await
     }
 
+    pub(crate) async fn get_write_multi_responses(
+        &mut self,
+        num_results: usize,
+    ) -> Result<(), Error> {
```

Comment:
We had to build this function instead of using the existing `get_read_write_response` function because `multi_set` behaves slightly differently on a protocol level.  The original function works with `get_many` because the command to set multiple values is simply `get key1 key2 ... keyn`, one request and one response.  For `multi_set` in the ASCII protocol, it is instead parsed by the memcached server as a series of individual set commands (e.g. `set key1/r/nvalue1/r/n set key2/r/nvalue 2/r/nEND/r/n`) and a response is issued to the buffer for each command.